kernel/src/engine/default/json.rs

```diff
@@ -81,15 +83,36 @@ impl<E: TaskExecutor> JsonHandler for DefaultJsonHandler<E> {
             return Ok(Box::new(std::iter::empty()));
         }
 
+        println!("Reading {} files", files.len());
+
         let schema: ArrowSchemaRef = Arc::new(physical_schema.as_ref().try_into()?);
         let file_opener = JsonOpener::new(self.batch_size, schema.clone(), self.store.clone());
-        FileStream::new_async_read_iterator(
-            self.task_executor.clone(),
-            schema,
-            Box::new(file_opener),
-            files,
-            self.readahead,
-        )
+
+        let (tx, rx) = mpsc::sync_channel(self.readahead);
```

Comment:
not sure we want to call this 'readahead' still - maybe just something like `buffer_size`?

Reply:
I have it controlling both the size of this channel and the number of futures which are buffered in the stream below